### PR TITLE
CLM-26523: include HPA notes

### PR DESCRIPTION
**Pull request by johnatan-velez**
Jira: https://sonatype.atlassian.net/browse/CLM-26523

### CODE REVIEW

File: chart/README.md [hunk 1]
```diff
@@ -541,14 +541,60 @@ Some example commands are shown below.
...
+command: `kubectl apply -f https://github.com/kubernetes-sigs/metrics-server/releases/latest/download/components.yaml`
+Also, defined resources limits for all the containers in the cluster are required for HPA to be able to properly compute
+metrics.
+
+### Examples
+
+Some example commands are shown below.
+
+   ```
+   helm install --namespace staging mycluster --dependency-update
```
Comment (mr-zepol) [inline chart/README.md:568]: should this example have the parameters above for hpa?

Comment (mr-zepol) [reply chart/README.md:568]: > commit: [df980a1](https://github.com/sonatype/nexus-iq-server-ha/commit/df980a17278b6efdcf4466bf54dc3e8a3503b7b3)

I'd add another example without the memory parameter, since from what I understand above, it's optional

File: chart/README.md [hunk 2]
```diff
@@ -541,14 +541,62 @@ Some example commands are shown below.
    sonatype/nexus-iq-server-ha --version <version>
    ```
 
+### HPA (Horizontal Pod Autoscaling)
```
Comment (kakumara) [inline chart/README.md:544]: I would just title this as `Autoscaling`

File: chart/README.md [hunk 3]
```diff
@@ -541,14 +541,62 @@ Some example commands are shown below.
...
+HPA is disabled by default. If you want to enable it, you need to set the `hpa.enabled` parameter to `true`, and set
+the `hpa.resources.memory.enabled` parameter to `true` to enable memory-based autoscaling.
+
+   ```
+   --set hpa.enabled=true
+   --set hpa.resources.memory.enabled=true
+   ```
+
+Note that HPA requires metrics-server to be installed in the cluster. If you are using EKS, you can install it by
+running the following
```
Comment (kakumara) [inline chart/README.md:558]: I don't think this is EKS specific, right?

File: chart/README.md [hunk 4]
```diff
@@ -541,14 +541,62 @@ Some example commands are shown below.
    sonatype/nexus-iq-server-ha --version <version>
    ```
 
+### HPA (Horizontal Pod Autoscaling)
+
+The HPA in Kubernetes automatically scales workload resources based on demand. It deploys more Pods
+for increased load and scales down if the load decreases. Implemented as a Kubernetes API resource and controller, it
+adjusts the target scale periodically based on observed metrics like CPU and memory utilization.
```
Comment (kakumara) [inline chart/README.md:548]: This sounds like a generic description of HPA itself which I'm not sure if its useful. Here's a suggestion specific to IQ

```
IQ HA helm chart includes support for Kubernetes Horizontal Pod Autoscaling (HPA). With this enabled you can set the cluster to automatically scale up/down based on the resource demand such as CPU and Memory utilization.

Note: When setting auto-scaling parameters please make sure to have sufficient hardware resources available in the underlying nodes meet the max pod demands. 
```

File: chart/README.md [hunk 5]
```diff
@@ -541,14 +541,62 @@ Some example commands are shown below.
...
+
+### Examples
+
+Some example commands are shown below.
+
+   ```
+   helm install --namespace staging mycluster --dependency-update
+    ...
+    --set hpa.enabled=true
+    --set hpa.resources.memory.enabled=tru
```
Comment (kakumara) [inline chart/README.md:571]: tru --> true

File: chart/README.md (hunk 4 above)
Comment (johnatan-velez) [reply chart/README.md]: I'm not sure if you were suggesting replace only the selected lines: 1a38b091427c7ff59c204622ba574423283ade9d

File: chart/README.md (hunk 3 above)
Comment (johnatan-velez) [reply chart/README.md]: Commit: 8524b9b2cf40a5f1a905c82307b9a25907a549e5

File: chart/README.md (hunk 2 above)
Comment (johnatan-velez) [reply chart/README.md]: commit; e3ed1e84384a7cf01707582318521ed9f87e7195

File: chart/README.md (hunk 1 above)
Comment (johnatan-velez) [reply chart/README.md:570]: They are not really optional. I mention that defined resources have to be set for HPA to be able to properly compute metrics. 
The optional would be just pass the parameters or update the `values.yml`

**Review by mr-zepol — APPROVED** [review]
+1 LGTM

File: chart/README.md [hunk 6]
```diff
@@ -541,14 +541,64 @@ Some example commands are shown below.
...
 
+### Autoscaling
+
+IQ HA helm chart includes support for Kubernetes Horizontal Pod Autoscaling (HPA). With this enabled you can set the
+cluster to automatically scale up/down based on the resource demand such as CPU and Memory utilization.
+
+Note: When setting auto-scaling parameters please make sure to have sufficient hardware resources available in the
+underlying nodes meet the max pod demands.
+HPA is disabled by default. If you want to enable it, you need to set the `hpa.enabled` parameter to `true`, and set
+the `hpa.resources.memory.enabled` parameter to `true` to enable memory-based autoscaling.
```
Comment (kakumara) [inline chart/README.md:552]: Lets remove the last part 
```
, and set
the `hpa.resources.memory.enabled` parameter to `true` to enable memory-based autoscaling.
```
as that is something we are describing in the table below and its default value

File: chart/README.md [hunk 7]
```diff
@@ -541,14 +541,64 @@ Some example commands are shown below.
...
+
+   ```
+   --set hpa.enabled=true
+   --set hpa.resources.memory.enabled=true
+   ```
```
Comment (kakumara) [inline chart/README.md:557]: Lets add a  **For example** pre-heading to this as the memory argument is optional and we don't want users to assume that is mandatory

File: chart/README.md [hunk 8]
```diff
@@ -541,14 +541,64 @@ Some example commands are shown below.
    sonatype/nexus-iq-server-ha --version <version>
    ```
 
+### Autoscaling
+
+IQ HA helm chart includes support for Kubernetes Horizontal Pod Autoscaling (HPA). With this enabled you can set the
```
Comment (kakumara) [inline chart/README.md:546]: We need to add something similar to **NEW FROM VERSION** to indicate the chart version this feature first enabled. 
Perhaps 
```
@Since version 166.0.0
```

File: chart/README.md [hunk 9]
```diff
@@ -541,14 +541,64 @@ Some example commands are shown below.
...
+    --set iq_server.resources.requests.memory="700M"
+    --set iq_server.resources.limits.memory="1024M"
+    --set forwarder.resources.requests.cpu="75m"
+    --set forwarder.resources.limits.cpu="200m"
+    --set forwarder.resources.requests.memory="100M"
+    --set forwarder.resources.limits.memory="200M"
+    --set aggregator.resources.requests.cpu="75m"
+    --set aggregator.resources.limits.cpu="200m"
+    --set aggregator.resources.requests.memory="100M"
+    --set aggregator.resources.limits.memory="200M" \
```
Comment (kakumara) [inline chart/README.md:585]: Does this needs setting? can't we default to some sensible values based on `hpa.enabled=true` in some way?

File: chart/README.md [hunk 10]
```diff
@@ -541,14 +541,64 @@ Some example commands are shown below.
...
+
+* Any PostgreSQL database, we recommend
+  one [setup for HA](https://www.postgresql.org/docs/current/high-availability.html)
+* Any Kubernetes cluster, we recommend a multi-node
+  cluster [setup for HA](https://kubernetes.io/docs/setup/production-environment/)
 * Any shared file system, we recommend a [Network File System (NFS)](https://en.wikipedia.org/wiki/Network_File_System)
 * Any [ingress controller](https://kubernetes.io/docs/concepts/services-networking/ingress-controllers/) pre-installed
-and configured in the cluster, we recommend the [ingress-nginx controller](https://github.com/kubernetes/ingress-nginx)
+  and configured in the cluster, we recommend
+  the [ingress-nginx controller](https://github.com/kubernetes/ingress-nginx)
```
Comment (kakumara) [inline chart/README.md:601]: unrelated changes

File: chart/README.md [hunk 11]
```diff
@@ -541,14 +541,63 @@ Some example commands are shown below.
    sonatype/nexus-iq-server-ha --version <version>
    ```
 
+### Autoscaling
+
+IQ HA helm chart includes support for Kubernetes Horizontal Pod Autoscaling (HPA). With this enabled you can set the
+cluster to automatically scale up/down based on the resource demand such as CPU and Memory utilization.
```
Comment (agustin-mrtz) [inline chart/README.md:547]: "up/down based on the resource demand such as CPU and Memory utilization." 
-> 
"up/down based on cpu and or memory utilization. "

File: chart/README.md [hunk 12]
```diff
@@ -541,14 +541,63 @@ Some example commands are shown below.
...
+
+Note: When setting auto-scaling parameters please make sure to have sufficient hardware resources available in the
+underlying nodes meet the max pod demands.
+HPA is disabled by default. If you want to enable it, you need to set the `hpa.enabled` parameter to `true`.
+
+   ```
+   --set hpa.enabled=true
+   --set hpa.resources.memory.enabled=true
```
Comment (agustin-mrtz) [inline chart/README.md:555]: --set hpa.resources.memory.enabled=true not needed

File: chart/README.md [hunk 13]
```diff
@@ -541,14 +541,63 @@ Some example commands are shown below.
...
    ```
 
+### Autoscaling
+
+IQ HA helm chart includes support for Kubernetes Horizontal Pod Autoscaling (HPA). With this enabled you can set the
+cluster to automatically scale up/down based on the resource demand such as CPU and Memory utilization.
+
+Note: When setting auto-scaling parameters please make sure to have sufficient hardware resources available in the
+underlying nodes meet the max pod demands.
+HPA is disabled by default. If you want to enable it, you need to set the `hpa.enabled` parameter to `true`.
```
Comment (agustin-mrtz) [inline chart/README.md:551]: need an explanation of default values

Comment (agustin-mrtz) [inline chart/README.md:551]: missing how to override utilization values and how to do so

Comment (agustin-mrtz) [reply chart/README.md:551]: specifically I mean that cpu is enabled by default and memory is now and they can disable either or based on preference

File: chart/README.md (hunk 9 above)
Comment (agustin-mrtz) [reply chart/README.md:584]: these are not needed - we need to update the iq-server-deployment template for the sidecar

File: chart/README.md (hunk 8 above)
Comment (johnatan-velez) [reply chart/README.md]: Commit: 095d334652f4c759547391f8d3d0f744be7cf230

File: chart/README.md (hunk 11 above)
Comment (johnatan-velez) [reply chart/README.md]: commit: c64cf6a87103364de113a7a8886112149f4b2556

File: chart/README.md (hunk 13 above)
Comment (johnatan-velez) [reply chart/README.md:551]: This is explained in the table below.

File: chart/README.md (hunk 12 above)
Comment (johnatan-velez) [reply chart/README.md]: commit: a5f2f4e88459cc6b5845096a1615dd6d2ead03a9

File: chart/README.md (hunk 7 above)
Comment (johnatan-velez) [reply chart/README.md]: Removed: commit: a5f2f4e88459cc6b5845096a1615dd6d2ead03a9

File: chart/README.md (hunk 9 above)
Comment (johnatan-velez) [reply chart/README.md]: IIUC this is the shortest param list for HPA to work. Commit: f775c04d6426d2fdd4167b8d95950293b154981a
@agustin-mrtz  please let me know it that is correct.

File: chart/README.md (hunk 10 above)
Comment (johnatan-velez) [reply chart/README.md]: Commit: 0ae7137d6c0abc3db925687ed63120452a67cd68

File: chart/README.md (hunk 13 above)
Comment (johnatan-velez) [reply chart/README.md:551]: What do you mean? The resource definition parameters are explained here: https://github.com/sonatype/nexus-iq-server-ha/blob/4412336e41844b2428b0ea0be86f6958fda0d815/chart/README.md?plain=1#L641C1-L644C44

Comment (kakumara) [reply chart/README.md:551]: May be let's add something like 

```
please refer to the "Chart Configuration Options" table below for detailed parameters for adjusting HPA configuration to match your needs.
```

File: chart/README.md [hunk 14]
```diff
@@ -541,6 +541,39 @@ Some example commands are shown below.
...
+
+   ```
+   --set hpa.enabled=true
+   ```
+
+HPA requires metrics-server to be installed in the cluster. You can install it by
+running the following
+command: `kubectl apply -f https://github.com/kubernetes-sigs/metrics-server/releases/latest/download/components.yaml`
```
Comment (kakumara) [inline chart/README.md:559]: with https://github.com/sonatype/nexus-iq-server-ha/pull/17 I think we might be able to modify this (or remove this)

File: chart/README.md (hunk 13 above)
Comment (johnatan-velez) [reply chart/README.md:551]: Commit. bd13d896fa8897391292795cda032af651c5ebdd

File: chart/README.md (hunk 14 above)
Comment (johnatan-velez) [reply chart/README.md]: Commit: 8fe7a63c8f6e03e20b757197730ca4f625ce777a

**Comment by johnatan-velez** [comment]
> 

If you refer to https://github.com/sonatype/nexus-iq-server-ha/pull/15 it is already merged.
